src/CheckBox/CheckBox.tsx

```diff
@@ -22,32 +23,52 @@ export const CheckBox: FC<CheckBoxProps> = ({
   toggle,
   error,
   errorMsg,
+  required = false,
 }) => {
   const id = useUniqueId(idProp)
+
   return (
     <>
       <BoxContainer id={id}>
-        <Text tag="span" typo="base" color={error ? 'error' : 'secondary'}>
-          {children}
-        </Text>
-
+        <LabelWrapper>
+          <Text tag="span" typo="base" color={getColor(error, required)}>
+            {children}
+          </Text>
+          {required && (
+            <Text tag="p" typo="base-small" color="error">
+              *
+            </Text>
+          )}
+        </LabelWrapper>
         <input type="checkbox" checked={checked} onChange={toggle} />
-        <Checkmark error={error} />
+        <Checkmark error={error} required={required} />
       </BoxContainer>
       {error && errorMsg && <ErrorBox>{errorMsg}</ErrorBox>}
     </>
   )
 }
 
-const Checkmark = styled.span<{ error?: boolean }>`
+const getColor = (error: boolean | undefined, required: boolean) => {
+  if (required && error) {
+    return 'error'
+  } else if (required) {
+    return 'secondary'
+  } else if (error) {
+    return 'error'
+  } else {
+    return 'secondary'
+  }
```

Comment:
isn't this the same as `((required && error) || error) ? 'error' : 'secondary'`?

Reply:
it is.